src/main/kotlin/pl/edu/agh/server/application/user/UserController.kt

```diff
@@ -1,11 +1,24 @@
 package pl.edu.agh.server.application.user
 
+import org.modelmapper.ModelMapper
+import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
-import pl.edu.agh.server.domain.user.UserRepository
+import pl.edu.agh.server.domain.dto.UserDTO
+import pl.edu.agh.server.domain.user.UserService
 
 @RestController
 @RequestMapping("/api/users")
 class UserController(
-    private val userRepository: UserRepository,
-)
+    private val userService: UserService,
+    private val modelMapper: ModelMapper,
+) {
+    @GetMapping
+    fun getUser(
+        @RequestParam(name = "email") email: String,
+    ): UserDTO {
+        val user = userService.getUserByEmail(email)
+        return modelMapper.map(user, UserDTO::class.java)
```

Comment:
IMO nie kazdy powininen miec dostep do endpointa z ktorego wyciaga sie dane o dowolnym uzytkowniku.

Widze ze uzywasz tego endpointu do zgarniecia danych o aktualnym uzytkowniku, wiec moze poprostu wyciagniesz email z requesta, to nie trzeba bedzie robic tu zadnego sprawdzania dostepu

Reply:
racja trafna uwaga